To: vim_dev@googlegroups.com
Subject: Patch 7.3.098
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.098
Problem:    Function that ignores error still causes called_emsg to be set.
	    E.g. when expand() fails the status line is disabled.
Solution:   Move check for emsg_not_now() up. (James Vega)
Files:	    src/message.c


*** ../vim-7.3.097/src/message.c	2010-10-20 21:22:17.000000000 +0200
--- src/message.c	2011-01-04 19:19:04.000000000 +0100
***************
*** 569,574 ****
--- 569,578 ----
      int		severe;
  #endif
  
+     /* Skip this if not giving error messages at the moment. */
+     if (emsg_not_now())
+ 	return TRUE;
+ 
      called_emsg = TRUE;
      ex_exitval = 1;
  
***************
*** 581,590 ****
      emsg_severe = FALSE;
  #endif
  
-     /* Skip this if not giving error messages at the moment. */
-     if (emsg_not_now())
- 	return TRUE;
- 
      if (!emsg_off || vim_strchr(p_debug, 't') != NULL)
      {
  #ifdef FEAT_EVAL
--- 585,590 ----
*** ../vim-7.3.097/src/version.c	2011-01-04 19:03:22.000000000 +0100
--- src/version.c	2011-01-04 19:24:30.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     98,
  /**/

-- 
Ed's Radiator Shop: The Best Place in Town to Take a Leak.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
